Runtime/Native/Android/NativeClient.cs

```diff
@@ -309,6 +310,71 @@ private void HandleNativeCrashes(IDictionary<string, string> backtraceAttributes
                         AndroidJNI.NewStringUTF(CrashType));
         }
 
+        private bool CanInitializeExecutableCrashHandler(String nativeLibraryDirectory, String handlerPath) {
+            return Directory.Exists(nativeLibraryDirectory) && File.Exists(handlerPath);
+        }
+
+        private bool InitializeExecutableCrashHandler(String minidumpUrl, String databasePath, String crashpadHandlerPath, IEnumerable<String> attachments) {
+            return Initialize(
+                AndroidJNI.NewStringUTF(minidumpUrl),
+                AndroidJNI.NewStringUTF(databasePath),
+                AndroidJNI.NewStringUTF(crashpadHandlerPath),
+                AndroidJNIHelper.ConvertToJNIArray(new string[0]),
+                AndroidJNIHelper.ConvertToJNIArray(new string[0]),
+                AndroidJNIHelper.ConvertToJNIArray(attachments.ToArray()),
+                _enableClientSideUnwinding,
+                (int)UnwindingMode);
+        }
+
+        private bool InitializeJavaCrashHandler(String minidumpUrl, String databasePath, String deviceAbi, String nativeDirectory, IEnumerable<String> attachments) {
+            if (String.IsNullOrEmpty(deviceAbi)) {
+                Debug.LogWarning("Cannot determine device ABI");
+                return false;
+            }
+
+            var envVariableDictionary =  Environment.GetEnvironmentVariables();
+            if (envVariableDictionary == null) {
+                Debug.LogWarning("Environment variables are not defined.");
+                return false;
+            }
+
+            // verify if the library is already extracted
+            var backtraceNativeLibraryPath = Path.Combine(nativeDirectory, _nativeLibraryName);
+            if (!File.Exists(backtraceNativeLibraryPath)) {
+                backtraceNativeLibraryPath = string.Format("{0}!/lib/{1}/{2}", Application.dataPath, deviceAbi, _nativeLibraryName);
```

Comment:
why not use `Path.Combine` here? Also, is the `!` required here?

Reply:
this is not a typical path - since `!` points to the path inside the apk. Thats why I wanted to use Format instead.